src/main/java/ca/serum390/godzilla/api/handlers/SalesHandler.java

```diff
@@ -3,38 +3,75 @@
+
+import ca.serum390.godzilla.data.repositories.SalesOrderRepository;
+import ca.serum390.godzilla.domain.sales.SalesOrder;
 import reactor.core.publisher.Mono;
 
 @Component
 public class SalesHandler {
 
-    private SalesHandler() {}
+    private final SalesOrderRepository salesOrders;
+
+    public SalesHandler(SalesOrderRepository salesOrder) {
+        this.salesOrders = salesOrder;
+    }
+
+    // Get All the sales orders
+    public Mono<ServerResponse> all(ServerRequest request) {
+        return ok().body(salesOrders.findAll(), SalesOrder.class);
+    }
 
-    public Mono<ServerResponse> demoSales(ServerRequest request) {
-        return ok().contentType(MediaType.APPLICATION_JSON)
-                   .bodyValue(buildDemoSalesList());
+    // Create a sales order
+    public Mono<ServerResponse> create(ServerRequest req) {
+        return req.bodyToMono(SalesOrder.class).flatMap(salesOrders::save).flatMap(id -> noContent().build());
     }
 
-    private static Map<Object, Object> buildDemoSalesList() {
-        return map().with("message", "Success. Here are your sales orders")
-                    .with("sales", Stream
-                        .iterate(1, i -> i <= 50, i -> i + 1)
-                        .map(val -> map()
-                            .with("id", val)
-                            .with("vendorName", "Vendor-" + val)
-                            .with("vendorAddress", "????")
-                            .with("vendorPhone", "123-456-7890")
-                            .with("salesOrders", List.of()))
-                        .collect(Collectors.toList()));
+    // Get a sales order
+    public Mono<ServerResponse> get(ServerRequest req) {
+        return salesOrders.findById(Integer.parseInt(req.pathVariable("id")))
+                .flatMap(salesOrder -> ok().body(Mono.just(salesOrder), SalesOrder.class))
+                .switchIfEmpty(notFound().build());
     }
+
+    // Delete a sales order
+    public Mono<ServerResponse> delete(ServerRequest req) {
+        return salesOrders.deleteById(Integer.parseInt(req.pathVariable("id"))).flatMap(deleted -> noContent().build());
+    }
+
+    // Update a sales order
+    public Mono<ServerResponse> update(ServerRequest req) {
+        var existed = salesOrders.findById(Integer.parseInt(req.pathVariable("id")));
+        return Mono.zip(data -> {
+            SalesOrder g = (SalesOrder) data[0];
+            SalesOrder g2 = (SalesOrder) data[1];
+            if (g2 != null) {
+                g.setCreatedDate(g2.getCreatedDate());
+                g.setDueDate(g2.getDueDate());
+                g.setDeliveryLocation(g2.getDeliveryLocation());
+                g.setOrderType(g2.getOrderType());
+            }
+            return g;
+        }, existed, req.bodyToMono(SalesOrder.class)).cast(SalesOrder.class)
+                .flatMap(SalesOrder -> salesOrders.update(SalesOrder.getCreatedDate(), SalesOrder.getDueDate(),
```

Comment:
Please rename this `SalesOrder` variable to `salesOrder`. 

Fyi, Java classes are usually written in SentenceCase (like your variable) so it is confusing to have variable names like that.

Better to use: camelCaseVariableNames